packages/costa/src/runnable.ts

```diff
@@ -193,9 +193,14 @@ export class PluginRunnable {
    * @param op
    * @param msg
    */
-  private send(op: PluginOperator, msg?: PluginMessage): boolean {
+  private send(op: PluginOperator, msg?: PluginMessage): Promise<void> {
     const data = PluginProtocol.stringify(op, msg);
-    return this.handle.send(data);
+    return new Promise((resolve, reject) => {
+      const rst = this.handle.send(data, (err) => {
+        err ? reject(err) : resolve();
+      });
+      rst || reject(new Error('subprocess send failed'));
```

Comment:
What does this mean?

Reply:
If `rst` is false, the callback will never been fired, so we need to reject an error by thin line.